bpf/lib/nat.h

```diff
@@ -372,6 +372,87 @@ snat_v4_rev_nat_handle_mapping(struct __ctx_buff *ctx,
 		       NAT_PUNT_TO_STACK : DROP_NAT_NO_MAPPING;
 }
 
+static __always_inline int snat_v4_icmp_rewrite_embedded(struct __ctx_buff *ctx,
+							 struct ipv4_ct_tuple *tuple,
+							 struct ipv4_nat_entry *state,
+							 __u32 off)
+{
+	struct csum_offset csum = {};
+	__be32 sum;
+	struct iphdr iphdr;
+	__u32 erroff;
+
+	if (ctx_load_bytes(ctx, off + sizeof(struct icmphdr), &iphdr,
+			   sizeof(iphdr)) < 0)
+		return DROP_INVALID;
+
+	/* This erroff is L4 offset of inner packet, where off is L4
+	 * offset of outer packet.
+	 *  off                     erroff
+	 *   [ ICMP ERROR | INNER IP | INNER L4 ...]
+	 */
```

Comment:
Let's just make that clear then, and do a `s/off/l4_off` and `s/erroff/inner_l4_off` ?

[I appreciate that you're making them `__u32` btw :+1:]

Reply:
Done